setup/static/render.js

```diff
@@ -35,80 +35,57 @@ $(document).ready(function() {
 
 $(document).ready(function() {
 	if ($("#database").val() == 'sqlite') {
-		$("#postgres_db").hide();
-	} else if ($("#database").val() == 'postgresql') {
-		$("#postgres_db").show();
-	} else if ($("#database").val() == 'mysql') {
-		$("#external_db").show();
-	}
-	if ($('#external_psql').prop('checked')) {
+		$("#external_db").hide();
+	} else if ($("#database").val() != 'sqlite') {
 		$("#external_db").show();
 	}
+
+	$("#webmail").click(function() {
+		if (this.value == 'roundcube') {
+			$("#db_flavor_rc_sel").show();
+		} else {
+			$("#db_flavor_rc_sel").hide();
+			$("#roundcube_db_user").prop('required',false);
+			$("#roundcube_db_pw").prop('required',false);
+			$("#roundcube_db_url").prop('required',false);
+			$("#roundcube_db_name").prop('required',false);
+		}
+	});
+
 	$("#database").click(function() {
 		if (this.value == 'sqlite') {
-			$("#postgres_db").hide();
-			$("#external_db").hide();
-		} else if (this.value == 'postgresql') {
-			$("#postgres_db").show();
 			$("#external_db").hide();
-			$("#external_psql").prop('checked', true);
-			$("#external_db").show();
-			$("#db_user").prop('required',true);
-			$("#db_pw").prop('required',true);
-			$("#db_url").prop('required',true);
-			$("#db_name").prop('required',true);
-			if ($("#webmail").val() == 'roundcube') {
-				$("#roundcube_external_db").show();
-				$("#roundcube_db_user").prop('required',true);
-				$("#roundcube_db_pw").prop('required',true);
-				$("#roundcube_db_url").prop('required',true);
-				$("#roundcube_db_name").prop('required',true);
-			}
-		} else if (this.value == 'mysql') {
-			$("#postgres_db").hide();
+			$("#db_user").prop('required',false);
+			$("#db_pw").prop('required',false);
+			$("#db_url").prop('required',false);
+			$("#db_name").prop('required',false);
+			$("#roundcube_db_user").prop('required',false);
+			$("#roundcube_db_pw").prop('required',false);
+			$("#roundcube_db_url").prop('required',false);
+			$("#roundcube_db_name").prop('required',false);
```

Comment:
why this? I think this should be removed.

Reply:
This is the fix for #1838. If these **required** flags are not set to false, you cannot click on Submit after switching back to SQLite.
The reason for this is that the hidden database form fields still expect all fields to be filled in  (they are still set to required,true). Setting the required property to false allows you to submit the form after switching the db flavour back to SQLite.